pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -729,29 +727,23 @@ func buildOutboundCatchAllNetworkFiltersOnly(push *model.PushContext, node *mode
 	return filterStack
 }
 
-// TODO: This code is still insufficient. Ideally we should be parsing all the virtual services
-// with TLS blocks and build the appropriate filter chain matches and routes here. And then finally
-// evaluate the left over unmatched TLS traffic using allow_any or registry_only.
-// See https://github.com/istio/istio/issues/21170
-func buildOutboundCatchAllNetworkFilterChains(_ *ConfigGeneratorImpl,
```

Comment:
Moved the logic from this function up to `buildVirtualOutboundListener` and refactored to move the catchall filter chain to `DefaultFilterChains`